### PR TITLE
Bump versions for guava to 26.0-jre and appengine-plugins-core 0.6.8

### PR DESCRIPTION
**Pull request by briandealwis**
Will require rebuilding your `ide-target-platform`.

### CODE REVIEW

**Comment by briandealwis** [comment]
I saw the 27.0-jre release, but the `listenablefuture` change seemed like it could be involved. But it looks benign.

**Comment by briandealwis** [comment]
We have to wait for `appengine-plugins-core` to move to 27.0-jre https://github.com/GoogleCloudPlatform/appengine-plugins-core/pull/726

**Comment by briandealwis** [comment]
Test failures are due to changes from https://github.com/GoogleCloudPlatform/appengine-plugins-core/pull/714 setting a default runtime.

**Comment by briandealwis** [comment]
PTAL: `AppEngineDescriptor.getRuntime()` now returns `java7` when the `<runtime>` element is not present.

**Comment by chanseokoh** [comment]
I see 0.6.6 in many `MANIFEST.MF` files in the `bump` branch.

```
(bump)$ git grep 0.6.6
plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/META-INF/MANIFEST.MF:Require-Bundle: com.google.cloud.tools.appengine;bundle-version="0.6.6",
plugins/com.google.cloud.tools.eclipse.appengine.deploy/META-INF/MANIFEST.MF:Require-Bundle: com.google.cloud.tools.appengine;bundle-version="0.6.6",
plugins/com.google.cloud.tools.eclipse.appengine.facets/META-INF/MANIFEST.MF:Require-Bundle: com.google.cloud.tools.appengine;bundle-version="0.6.6",
plugins/com.google.cloud.tools.eclipse.appengine.libraries/META-INF/MANIFEST.MF:Import-Package: com.google.cloud.tools.appengine.cloudsdk;version="0.6.6",
plugins/com.google.cloud.tools.eclipse.appengine.libraries/META-INF/MANIFEST.MF: com.google.cloud.tools.libraries;version="0.6.6",
plugins/com.google.cloud.tools.eclipse.appengine.libraries/META-INF/MANIFEST.MF: com.google.cloud.tools.libraries.json;version="0.6.6",
plugins/com.google.cloud.tools.eclipse.appengine.localserver/META-INF/MANIFEST.MF:Require-Bundle: com.google.cloud.tools.appengine;bundle-version="0.6.6",
plugins/com.google.cloud.tools.eclipse.appengine.newproject/META-INF/MANIFEST.MF:Require-Bundle: com.google.cloud.tools.appengine;bundle-version="0.6.6",
plugins/com.google.cloud.tools.eclipse.appengine.standard.java8.test/META-INF/MANIFEST.MF:Import-Package: com.google.cloud.tools.appengine;version="0.6.6",
plugins/com.google.cloud.tools.eclipse.appengine.standard.java8/META-INF/MANIFEST.MF:Import-Package: com.google.cloud.tools.appengine;version="0.6.6",
plugins/com.google.cloud.tools.eclipse.appengine.standard.java8/META-INF/MANIFEST.MF: com.google.cloud.tools.appengine.api;version="0.6.6",
plugins/com.google.cloud.tools.eclipse.bugreport.test/META-INF/MANIFEST.MF:Import-Package: com.google.cloud.tools.appengine.cloudsdk.serialization;version="0.6.6",
plugins/com.google.cloud.tools.eclipse.bugreport/META-INF/MANIFEST.MF:Import-Package: com.google.cloud.tools.appengine.api;version="0.6.6",
plugins/com.google.cloud.tools.eclipse.bugreport/META-INF/MANIFEST.MF: com.google.cloud.tools.appengine.cloudsdk;version="0.6.6",
plugins/com.google.cloud.tools.eclipse.bugreport/META-INF/MANIFEST.MF: com.google.cloud.tools.appengine.cloudsdk.serialization;version="0.6.6",
plugins/com.google.cloud.tools.eclipse.integration.appengine/META-INF/MANIFEST.MF:Import-Package: com.google.cloud.tools.appengine.cloudsdk;version="0.6.6",
plugins/com.google.cloud.tools.eclipse.sdk.ui.test/META-INF/MANIFEST.MF:Import-Package: com.google.cloud.tools.appengine.cloudsdk.serialization;version="0.6.6",
plugins/com.google.cloud.tools.eclipse.sdk.ui/META-INF/MANIFEST.MF:Import-Package: com.google.cloud.tools.appengine;bundle-version="0.6.6",
plugins/com.google.cloud.tools.eclipse.sdk/META-INF/MANIFEST.MF:Require-Bundle: com.google.cloud.tools.appengine;bundle-version="0.6.6"
```

File: plugins/com.google.cloud.tools.eclipse.appengine.facets.test/src/com/google/cloud/tools/eclipse/appengine/facets/StandardFacetInstallDelegateTest.java [hunk 1]
```diff
@@ -60,7 +60,8 @@ public void testCreateConfigFiles_appengineWebXml()
 
     try (InputStream in = appengineWebXml.getContents(true)) {
       AppEngineDescriptor descriptor = AppEngineDescriptor.parse(in);
-      assertNull(descriptor.getRuntime());
+      // AppEngineDescriptor now treats an empty runtime as java7 as of 0.6.
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.facets.test/src/com/google/cloud/tools/eclipse/appengine/facets/StandardFacetInstallDelegateTest.java]: nit: "now" is redundant

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -9,9 +9,9 @@ Bundle-RequiredExecutionEnvironment: JavaSE-1.8
 Bundle-Activator: com.google.cloud.tools.eclipse.appengine.localserver.Activator
 Bundle-ActivationPolicy: lazy
 Bundle-Localization: plugin
-Require-Bundle: com.google.cloud.tools.appengine;bundle-version="0.6.6",
+Require-Bundle: com.google.cloud.tools.appengine;bundle-version="[0.6.8,0.7.0)",
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver/META-INF/MANIFEST.MF]: Does specifying 0.6.8 not prevent higher versions?

File: pom.xml [hunk 3]
```diff
@@ -108,13 +108,13 @@
       -->
       <groupId>org.apache.commons</groupId>
       <artifactId>commons-compress</artifactId>
-      <version>1.17</version>
+      <version>1.18</version>
```
Comment (elharo) [inline pom.xml]: In appengine-plugins-core 0.6.8 this is still 1.17. It changed to 1.18 in 0.7.0. Will this be a conflict?

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver/META-INF/MANIFEST.MF (hunk 2 above)
Comment (briandealwis) [reply plugins/com.google.cloud.tools.eclipse.appengine.localserver/META-INF/MANIFEST.MF]: It's the equivalent of "[0.6.8,∞]"

File: pom.xml (hunk 3 above)
Comment (briandealwis) [reply pom.xml]: `appengine-plugins-core`'s MANIFEST.MF specifies:
```
Import-Package: ...
  org.apache.commons.compress.archivers.tar;version="[1.17,2)",
  org.apache.commons.compress.archivers.zip;version="[1.17,2)",
  org.apache.commons.compress.compressors.gzip;version="[1.17,2)",
  org.apache.commons.compress.utils;version="[1.17,2)",
```
So we're good.

**Pull request merged**